.github/workflows/ruby.yml

```diff
@@ -85,10 +85,55 @@ jobs:
           }
 
       - name: test
-        timeout-minutes: 10
+        timeout-minutes: 8
         run:  |
           # RubyGems < 3, no frozen strings
           if ('${{ matrix.ruby }}' -ge '2.6') {
             $env:RUBYOPT = '--enable=frozen-string-literal'
           }
           bundle exec rake
+
+  nonMRIHead:
+    name: >-
+      ${{ matrix.cfg.os }}, ${{ matrix.cfg.ruby }}
+    env:
+      CI: true
+      TESTOPTS: -v
+
+    runs-on: ${{ matrix.cfg.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        cfg:
+          - { os: ubuntu-latest, ruby: jruby-head       }
+          - { os: ubuntu-latest, ruby: truffleruby-head }
+
+    steps:
+      - name: repo checkout
+        uses: actions/checkout@v2
+
+      - name: load ruby, ragel
+        uses: MSP-Greg/setup-ruby-pkgs@v1
+        with:
+          ruby-version: ${{ matrix.cfg.ruby }}
+          apt-get: ragel
+          brew: ragel
+
+      - name: bundle install
+        run:  |
+          bundle install --jobs 4 --retry 3 --path=.bundle/puma
+
+      - name: compile
+        continue-on-error: true
+        run:  bundle exec rake compile
+
+      - name: test
+        timeout-minutes: 8
+        continue-on-error: true
+        if: success()
+        run:  |
+          # RubyGems < 3, no frozen strings
+          if [[ "2.6" < "${{ matrix.cfg.ruby }}.0" ]]; then
```

Comment:
This check won't work, since there is no version in the matrix entries above

Reply:
Please check the log of the 'load ruby, ragel' step, and you'll see the correct matrix.cfg.ruby version loaded from setup-ruby's log.  This is another supported way to construct a matrix.

Reply:
It results in checks like:
```
# RubyGems < 3, no frozen strings
  if [[ "2.6" < "jruby-head.0" ]]; then
    export RUBYOPT="--enable=frozen-string-literal"
  fi
```
which don't make much sense (https://github.com/puma/puma/pull/2185/checks?check_run_id=509502033).

Reply:
It was my understanding that jruby-head supported `--enable=frozen-string-literal`.  Is that correct?  As I thought the conditional is true...

Reply:
I don't really care as long it evals correctly FWIW

Reply:
I'm not seeing frozen string errors...

Reply:
Since this job applies only for jruby and truffleruby, how about simply removing the conditional and always `--enable=frozen-string-literal`?